cabal-install/Distribution/Client/CmdInstall/Utils.hs

```diff
@@ -0,0 +1,41 @@
+module Distribution.Client.CmdInstall.Utils
+  ( reportCannotPruneDependencies
+  , warnIfNoExes
+  ) where
+
+import Prelude ()
+import Distribution.Client.Compat.Prelude
+
+import qualified Data.Map as Map
+
+import Distribution.Client.CmdErrorMessages
+       ( plural, listPlural, renderCannotPruneDependencies, showTargetSelector )
+import Distribution.Client.ProjectOrchestration
+       ( CannotPruneDependencies(..), ComponentName(..)
+       , ComponentTarget(..) , ProjectBuildContext(..) )
+import Distribution.Simple.Utils
+       ( die', warn )
+import Distribution.Verbosity
+       ( Verbosity )
+
+-- | Emits a warning if the build context does not contain any executables.
+warnIfNoExes :: Verbosity -> ProjectBuildContext -> IO ()
```

Comment:
I think this should be just in `CmdInstall` implementation. It's used only there.

Reply:
See comment See top level review comment.

Reply:
TL;DR; a lot of the other "implementation detail" code will be moving to this module, but I also realize this module should likely be named `Internal` instead of `Utils`.